lib/connectors/memory.js

```diff
@@ -351,26 +351,54 @@ Memory.prototype.fromDb = function(model, data) {
   if (!data) return null;
   data = deserialize(data);
   var props = this._models[model].properties;
+  function cast(prop, val, props) {
```

Comment:
I can do this, but keep in mind that the only reason I'm moving this logic into its own function is so that it can call itself for the nested properties. The logic is still unique to the `fromDb` method and won't be used elsewhere.